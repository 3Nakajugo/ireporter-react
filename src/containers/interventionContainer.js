import React, { Component } from 'react';
import { connect } from 'react-redux';
import { interventionAction } from '../redux/actions/interventionAction';
import  InterventionComponent  from '../components/Intervention'


export class InterventionContainer extends Component {
  componentDidMount() {
    this.props.interventionAction();
  }
  render() {
    return (
      <div>
        <InterventionComponent state={this.props} />
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  interventions: state.interventions.interventionResponse
});

export default connect(mapStateToProps, { interventionAction })(InterventionContainer);
