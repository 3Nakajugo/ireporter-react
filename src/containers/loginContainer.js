import React, { Component } from 'react'
import { connect } from 'react-redux';
import { loginAction } from '../redux/actions/loginAction';
import LoginComponent from '../components/Login/login'

export class LoginContainer extends Component {
  constructor(props) {
    super(props);
    this.state = {
      user_name: '',
      password: ''
    };
  }

  onSubmit = (e) => {
    e.preventDefault();
    const { user_name, password } = this.state;
    const data = {
      user_name: user_name,
      password: password
    }
    this.props.loginAction(data);
  }

  onChange = (e) => {
    this.setState({ [e.target.name]: e.target.value });
  }
  render() {
    return (
      <div>
        <LoginComponent
          onChange={this.onChange}
          onSubmit={this.onSubmit}
        />
      </div>
    )
  }
}

export const mapStateToProps = state => ({
  login: state.login.loginResponse
});

export default connect(
  mapStateToProps,
  { loginAction },
)(LoginContainer);
