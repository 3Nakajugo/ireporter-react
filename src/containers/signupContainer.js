import React, { Component } from 'react';
import { connect } from 'react-redux';
import { signupAction } from '../redux/actions/signupActions';
import SignupComponent from '../components/Signup/Signup';

export class SignupContainer extends Component {
  constructor(props) {
    super(props);
    this.state = {
      first_name: '',
      last_name: '',
      telephone: '',
      user_name: '',
      email: '',
      password: '',
    };
  }

  onSubmit = (e) => {
    e.preventDefault();
    const {
      first_name, last_name, telephone, user_name, email, password
    } = this.state;
    const data = {
      first_name:first_name,
      last_name:last_name,
      telephone:telephone,
      user_name:user_name,
      email:email,
      password:password
    }
    this.props.signupAction(data);
  }

  onChange = (e) => {
    this.setState({ [e.target.name]: e.target.value });
  }
  render(){
    const { errors } = this.props;
    return (
      <div>
        <SignupComponent
          errors={errors}
          onChange={this.onChange}
          onSubmit={this.onSubmit}
        />
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  signup: state.signup.signupResponse,
  error: state.signup.errors
});

export default connect(
  mapStateToProps,
  { signupAction },
)(SignupContainer);
