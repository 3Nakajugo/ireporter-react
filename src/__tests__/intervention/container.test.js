import React from 'react';
import { shallow } from "enzyme";
import { InterventionContainer, mapStateToProps } from "../../containers/interventionContainer";

describe('tests intervention container', () => {
  let wrapper;
  const props = {
    interventionAction: jest.fn()
  };
  beforeEach(() => {
    wrapper = shallow(<InterventionContainer {...props} />);
  });

  it("should render without crashing", () => {
    expect(wrapper).toMatchSnapshot();
  });


  it("should map state to props", () => {
    const initialState = {
      interventions: {
        errors: {},
        interventionResponse: {},
      }
    };
    expect(mapStateToProps(initialState)).toEqual({
      interventions: {}
    });
  });

})
