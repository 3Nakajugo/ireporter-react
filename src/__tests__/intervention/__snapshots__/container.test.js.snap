// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests intervention container should render without crashing 1`] = `ShallowWrapper {}`;
